reference/pipeline/expressions/index.md

```diff
@@ -0,0 +1,214 @@
+expression `${new java.text.SimpleDateFormat("MM-DD-YYYY").format(new
+java.util.Date())}`.
+
+Similarly, you can call static methods using the syntax
+`T(fully.qualified.class.name).methodName()`. For example, to calculate the date
+5 days from now, you can call:
+`${T(java.time.LocalDate).now().plusDays(5).toString()}`.
+
+### Comparisons
+
+You can use relational operators to compare values in an expression, such as
+`${instance["size"] > 400}` or `${parameters["runCanary"] == "true"}`.
+
+Note that you may need to transform some values for the comparisons to work
+properly. In the example above, `parameters["runCanary"]` returns a string
+rather than a boolean. To use it in a comparision, you need to either compare
+it to a string or convert it to a boolean:
+`${#toBoolean(parameters["runCanary"]) == true}`.
+
+Another example is that the status attribute of a stage is actually an enum
+internally, not a string. To compare the status to a string, you need to call
+`.toString()` on the result. For example:
+`${#stage("Deploy")["status"].toString() == "SUCCEEDED"}`.
+
+### Functions
+
+The expression language has several built in helper functions that simplify
+common use cases. The syntax for calling built-in functions is
+`#functionName(params)`. For example, `#fromUrl("http://www.netflix.com")`. See
+the full list of functions [here](#helper-functions).
+
+### Lists
+
+You can also use lists in your expressions. Spinnaker always provides a list
+called _stages_, which you can use to access each stage by index:
+`${execution["stages"][0]}` returns the value of the first stage in your
+pipeline.  Keep in mind, though, that if you change the order in which your
+stages execute in the pipeline (separately from your expression), the value of
+`stages[i]` will also change.
+
+### Maps
+
+You can use maps to access data from the JSON representation of your pipeline.
+For example, `${trigger["properties"]["example"]}` evaluates to the value of
+the `example` trigger property. To see the available values for a given
+pipeline, view the execution JSON:
+1. Go to an execution of your pipeline.
+2. Click on _Details_ to expand the pipeline.
+3. Click on the source link under the pipeline execution, which opens a new tab
+    containing the JSON details of your pipeline execution.
+    _[[screenshot](https://screenshot.googleplex.com/U1rVRjdiqKC.png)]_
+
+Note that while Spinnaker supports both dot (`map.value`) and square bracket
+(`map["value"]`) notation, we recommend using square brackets. That is, prefer
+`trigger["properties"]["value"]` instead of `trigger.properties.value`. There
+are a few places where using dot notation produces unexpected results, such as
+after a filter operation or when getting nested JSON values from an URL.
+
+You can filter maps using `.?`. For example, to filter a list of stages by type,
+you would use `${execution["stages"].?[type == "bake"]}`.
```

Comment:
The result of such filtering is a list, so to access an element it should be referenced as such...